src/test/kotlin/com/terraformation/backend/funder/FundingEntityServiceTest.kt

```diff
@@ -0,0 +1,306 @@
+import com.terraformation.backend.db.DatabaseTest
+import com.terraformation.backend.db.funder.FundingEntityId
+import com.terraformation.backend.db.funder.tables.pojos.FundingEntitiesRow
+import com.terraformation.backend.db.funder.tables.records.FundingEntitiesRecord
+import com.terraformation.backend.db.funder.tables.records.FundingEntityProjectsRecord
+import com.terraformation.backend.db.funder.tables.references.FUNDING_ENTITIES
+import com.terraformation.backend.db.funder.tables.references.FUNDING_ENTITY_PROJECTS
+import com.terraformation.backend.db.funder.tables.references.FUNDING_ENTITY_USERS
+import com.terraformation.backend.funder.db.FundingEntityExistsException
+import com.terraformation.backend.funder.db.FundingEntityNotFoundException
+import com.terraformation.backend.funder.db.FundingEntityStore
+import com.terraformation.backend.funder.db.FundingEntityUserStore
+import com.terraformation.backend.mockUser
+import io.mockk.every
+import java.time.Instant
+import java.util.UUID
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.security.access.AccessDeniedException
+
+class FundingEntityServiceTest : DatabaseTest(), RunsAsUser {
+  override val user: TerrawareUser = mockUser()
+
+  private val clock = TestClock()
+  private val fundingEntityStore by lazy { FundingEntityStore(dslContext) }
+  private val fundingEntityUserStore by lazy { FundingEntityUserStore(dslContext) }
+  private val service by lazy {
+    FundingEntityService(clock, dslContext, fundingEntityStore, fundingEntityUserStore)
+  }
+
+  private val fundingEntityId by lazy { insertFundingEntity() }
+  private lateinit var fundingEntityName: String
+
+  @BeforeEach
+  fun setUp() {
+    insertOrganization()
+
+    every { user.canReadFundingEntities() } returns true
+    every { user.canCreateFundingEntities() } returns true
+    every { user.canDeleteFundingEntities() } returns true
+    every { user.canUpdateFundingEntities() } returns true
+    every { user.canUpdateFundingEntityProjects() } returns true
+
+    // has to be after permissions in order to read correctly
+    fundingEntityName = fundingEntityStore.fetchOneById(fundingEntityId).name
+  }
+
+  @Test
+  fun `create requires user to be able to manage funding entities`() {
+    every { user.canCreateFundingEntities() } returns false
+
+    assertThrows<AccessDeniedException> { service.create("Some Other Entity") }
+  }
+
+  @Test
+  fun `create populates created and modified fields`() {
+    val initialRecord =
+        FundingEntitiesRecord(
+            createdBy = currentUser().userId,
+            createdTime = clock.instant(),
+            id = fundingEntityId,
+            modifiedBy = currentUser().userId,
+            modifiedTime = clock.instant(),
+            name = fundingEntityName,
+        )
```

Comment:
A few other options here.

1. If you make `fundingEntityName` lazy too, then you don't need `initialRecord` because this test doesn't refer to the lazy-inserted initial row so it just won't get inserted at all.
2. It seems like a decent percentage of the tests in this file don't use the initial record. Maybe it shouldn't be a test-class-level fixture at all, but rather a thing each test inserts if needed. In that case you don't need `initialRecord` either.
3. If you _do_ want the initial row to always be present, then rather than constructing its record like this, you could do something like `dslContext.selectFrom(FUNDING_ENTITIES).fetchOne()`.

Reply:
Good thought, I went with option 2.